tests/e2e_test.py

```diff
@@ -138,3 +151,27 @@ def test_target_1(setup):
     finally:
         khstz.kill()
         delete_target(f"{target_name}.yaml")
+
+
+def test_target_2(setup):
```

Comment:
Could we `pytest.mark.parametrize` this as the tests are practically the same?

we could `wait_deployment_scale(*, name: str, replicas: int)` -> `wait_scale(*, kind: str, name: str, replicas: int)`

And pass `deployment` or `statefulset` as a parameter as well.